Tools/i18n/pygettext.py

```diff
@@ -373,7 +373,42 @@ def parse_spec(spec):
         raise ValueError(f'Invalid keyword spec {spec!r}: '
                          'msgctxt cannot appear without msgid')
 
-    return name, {v: k for k, v in result.items()}
+    return name, result
+
+
+def unparse_spec(name, spec):
+    """Unparse a keyword spec dictionary into a string."""
+    if spec == {'msgid': 0}:
+        return name
+
+    parts = []
+    for arg, pos in sorted(spec.items(), key=lambda x: x[1]):
+        if arg == 'msgctxt':
+            parts.append(f'{pos + 1}c')
+        else:
+            parts.append(str(pos + 1))
+    return f'{name}:{','.join(parts)}'
+
+
+def process_keywords(keywords, *, no_default_keywords):
+    custom_keywords_list = [parse_spec(spec) for
+                            spec in dict.fromkeys(keywords)]
+    custom_keywords = {}
+    for name, spec in custom_keywords_list:
```

Comment:
I think this can be simpler:
```suggestion
    custom_keywords = {}
    for spec in dict.fromkeys(keywords):
        name, spec = parse_spec(spec)
```